grant/projects/offchain-subnames.md

```diff
@@ -0,0 +1,87 @@
+
+<!-- Key features of the project include:
+1. Offchain storage of subname-address mappings on Darwinia Chain
+2. Integration with ENS on Ethereum using EIP-3668 for secure offchain data retrieval
+3. A gateway service to facilitate communication between Ethereum and Darwinia Chain
+4. User-friendly UIs for subname registration, update, and resolution.
+5. Admin UI for subname management. -->
+
+This project addresses the need for efficient subdomain management within the Darwinia ecosystem, as outlined in the RFP (https://github.com/ringecosystem/collaboration/issues/61), and demonstrates the potential for cross-chain interoperability in domain name services.
+
+### This project will benefit the Darwinia ecosystem in the following ways:
+
+1. It will provide a more efficient and cost-effective way to manage subdomains on the Darwinia Chain, reducing the need for users to pay high gas fees on Ethereum.
+2. It will enhance the usability of the 'ringdao.eth' domain by allowing users to register, update, and resolve subnames with reduced costs and improved performance.
+3. It will demonstrate the potential for cross-chain interoperability in domain name services, opening up new possibilities for collaboration and innovation within the Darwinia ecosystem.
+
+## Team
+
+- Aki Wu([wuminzhe](https://github.com/wuminzhe))  
+  A full stack developer focused on the web3 space. He has been working on darwinia ecosystem for a long time.
+
+- Echo([Echo](https://github.com/hujw77))  
+  Contracts developer from Itering.
+
+
+## Team Code Repos
+
+- https://github.com/wuminzhe/basenames
+  The repo for the basenames contracts.
+
+- https://github.com/wuminzhe/offchain-resolver  
+  The repo provides the offchain resolve client.
+
+- https://github.com/wuminzhe/subnames-ui
+  The repo for the UI.
+
+- https://github.com/wuminzhe/subnames-gateway
+  The repo for the offchain gateway.
+
+## Development Roadmap
+
+### Milestone 1
+
+Estimated Duration: 8 weeks
+
+Cost: 1 ETH
+
+Address: 0x2Da8ccfe0dD2165B8d939eaBf4E3697C4Adb6FDd
+
+#### Tasks:
+
+1. Contracts
+   1. Pre research.  
+      Echo did and will continue the research.   
+      Aki registered a real basename([akiwu.base.eth](https://www.base.org/name/akiwu)) on the https://www.base.org/names to verify the operations logic.
+   2. Fork basenames repo and update the contracts to align with Darwinia Chain.
+   3. Deploy to testnet(Dariwnia Koi).
+2. Gateway  
+   1. Implement the EIP-3668 offchain gateway.
+   2. Deployment.
+3. UI
+    1. Implement logic for
+       1. wallet connect including customizing it to support subname.
+       2. register including reverse register and fee calculation
+       3. renew
+    2. Deploy to testnet(Darwinia Koi).
+4. Production deployment
+
```

Comment:
Please refine the Production Development section, clarifying that it will be deployed on the Darwinia chain, and list the key deliverables in detail:

	1.	Scope of the open-source code repository, including its license.
	2.	Documentation and testing (if available).
	3.	Pricing plan, including the arrangement and distribution of revenue.

Reply:
yes, I will update them.